tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,861 @@
+);
+
+CREATE TYPE IF NOT EXISTS ArmorClass AS ENUM (
+	'Light', 
+	'Medium', 
+	'Heavy'
+);
+
+CREATE TYPE IF NOT EXISTS EquipmentSlot AS ENUM (
+	'None',
+	'Armor', 
+	'Belts', 
+	'Body', 
+	'Chest', 
+	'Eyes', 
+	'Feet', 
+	'Hands', 
+	'Head', 
+	'Headband', 
+	'Neck', 
+	'Ring (left)', 
+	'Ring (right)', 
+	'Shield', 
+	'Shoulders', 
+	'Wrist'
+);
+
+CREATE TYPE IF NOT EXISTS DamageType AS ENUM (
+	'Bludgeoning', 
+	'Slashing', 
+	'Piercing', 
+	'Energy', 
+	'Acid', 
+	'Fire', 
+	'Electricity', 
+	'Cold', 
+	'Sonic', 
+	'Positive', 
+	'Negative', 
+	'Nonlethal'
+);
+
+--table declarations
+--user creds table - also used for hashing info
+CREATE TABLE IF NOT EXISTS Users (
+	id     			UUID PRIMARY KEY,
+
+	email			TEXT 
+		NOT NULL,
+        username       		TEXT 
+		NOT NULL,
+	nickname		TEXT 
+		NOT NULL,
+        is_admin    		BOOL 
+		NOT NULL,
+
+        pass_hash   		BYTEA 
+		NOT NULL,
+        salt        		BYTEA 
+		NOT NULL,
+
+        time_cost   		INT 
+		NOT NULL,
+        memory      		INT 
+		NOT NULL,
+        threads     		INT 
+		NOT NULL
+);
+
+--Character overview table
+CREATE TABLE IF NOT EXISTS Characters (
+        char_id         	UUID PRIMARY KEY,
+
+	id			UUID REFERENCES Users 
+		NOT NULL,
+        race_id         	UUID REFERENCES Races 
+		NOT NULL,
+        deity_id        	UUID REFERENCES Deities 
+		NOT NULL,
+
+        name            	TEXT 
+		NOT NULL,
+        age             	INT 
+		NOT NULL 
+		CHECK (age > 0),
+        gender          	Gender,
+        alignment       	Alignment 
+		NOT NULL,
+        backstory       	TEXT,
+        height          	INT,
+        weight          	INT,
+        size            	Size 
+		NOT NULL,
+
+        strength        	INT 
+		NOT NULL 
+		CHECK (strength >= 0 AND strength <= 20),
+        dexterity       	INT 
+		NOT NULL 
+		CHECK (dexterity >= 0 AND dexterity <= 20),
+        constitution    	INT 
+		NOT NULL
+		CHECK (constitution >= 0 AND constitution <= 20),
+        intelligence    	INT 
+		NOT NULL
+		CHECK (intelligence >= 0 AND intelligence <= 20),
+        wisdom          	INT
+		NOT NULL
+		CHECK (wisdom >= 0 AND wisdom <= 20),
+        charisma        	INT
+		NOT NULL
+		CHECK (charisma >= 0 AND charisma <= 20),
+
+        max_hp          	INT
+		NOT NULL
+		CHECK (max_hp >= 0),
+        damage          	INT
+		NOT NULL
+		CHECK (damage >= 0),
+        nonlethal       	INT
+		NOT NULL
+		CHECK (nonlethal >= 0),
+
+        copper          	INT
+		NOT NULL
+		CHECK (copper >= 0),
+        silver          	INT
+		NOT NULL
+		CHECK (silver >= 0),
+        gold            	INT
+		NOT NULL
+		CHECK (gold >= 0),
+        platinum        	INT
+		NOT NULL
+		CHECK (platinum >= 0)
+);
+
+--Race tables
+CREATE TABLE IF NOT EXISTS Races (
+	race_id 		UUID PRIMARY KEY,
+	type_id			UUID REFERENCES RaceTypes
+		NOT NULL,
+	subtype_id		UUID REFERENCES RaceSubtypes,
+
+	name			TEXT
+		NOT NULL,
+	move_speed		INT
+		NOT NULL,
+	size			Size
+		NOT NULL,
+	languages		TEXT
```

Comment:
If we have a place where languages are defined, this should either be an array of those (I know nothing about that feature) or this should be tied to a junction table so we can easily determine how many languages are available to the race and which ones.

Reply:
Implemented as an array. They work like you'd expect, being an array of values. I'm not sure how joins work on this, but I also am not designing it with that in mind. As far as I'm designing the database, array types are only for data endpoints, never keys.

Reply:
>  I'm not sure how joins work on this, but I also am not designing it with that in mind.
I think I would prefer a junction table, then, because we know that we can do `SELECT`  queries and `JOIN`s using those. If you can find that it's just as easy to do with arrays, I'm fine with keeping it, though.